### ISSUE TITLE
incorporate parquet files?

### ISSUE THREAD

**Issue by wibeasley** [IQSS/dataverse#9897]
Has there been any discussion of using [parquet files](https://www.databricks.com/glossary/what-is-parquet) at some level of dataverse?  (I see it mentioned in only [one issue](https://github.com/IQSS/dataverse/issues/6807#issuecomment-1398799794).)
> Apache Parquet is an open source, column-oriented data file format designed for efficient data storage and retrieval. It provides efficient data compression and encoding schemes with enhanced performance to handle complex data in bulk. Apache Parquet is designed to be a common interchange format for both batch and interactive workloads.

I've used them some, and I love how they work well with [R](https://r4ds.hadley.nz/arrow.html#sec-parquet), [Python](https://saturncloud.io/blog/how-to-write-data-to-parquet-with-python/), [DuckDb](https://duckdb.org/docs/data/parquet/overview.html), [Spark](https://sparkbyexamples.com/spark/spark-read-write-dataframe-parquet-example/), and others.

Several R programmers (like @kuriwaki) have advocated for [rds files over RData files](https://github.com/IQSS/dataverse-client-r/issues/35#issuecomment-567752925).  From my recent experience with parquet files, they have all the advertised advantages of rds files (eg, compression, strong-typing, and factor levels), plus the appeal of interoperability with other platforms.

I haven't thought much beyond this.  But when I read about [problems with RData files](https://github.com/IQSS/dataverse-client-r/issues/127) and the [messiness of Rserve](https://github.com/IQSS/dataverse/issues/9490#issuecomment-1492640510) described by @landreev, I see parquet as a improvement for many reasons --not least is the ability to replace a flaky remote instance with a local parquet library.

cc: @pdurbin

**Comment by pdurbin** [comment]
Yes! Recently 2020 data from the US Census was published in Harvard Dataverse in parquet format: https://dataverse.harvard.edu/dataset.xhtml?persistentId=doi:10.7910/DVN/5LAVKV

(2010 data was published as well: https://dataverse.harvard.edu/dataset.xhtml?persistentId=doi:10.7910/DVN/1OR2A6 )

This work is with the US Census is ongoing and being tracked here:

- https://github.com/IQSS/dataverse.harvard.edu/issues/218

That said, no, Dataverse doesn't have any particular support for parquet files. In the examples above the parquet files are in a zip file. Here's a preview of the 2020 zip:

![Screen Shot 2023-09-08 at 3 49 20 PM](https://github.com/IQSS/dataverse/assets/21006/a6205f67-2aba-4d1d-b203-7bbaa1f59a49)

**Comment by kuriwaki** [comment]
One note here is that dataverse does not seem to "Unzip" a compressed parquet collection in a way that respects the file hierarchy.  In this [example](https://photos.app.goo.gl/bChV9NiXXo5bw25JA) I just made it says it "failed to unzip the file properly". The file itself is still intact: the user can unzip it themselves after downloading, and Phil's screenshot above shows that the file hierarchy can be viewed as metadata, which may be the best way forward. But just a note.

**Comment by kuriwaki** [comment]
The parquet I'm working with now _does_ expand out the partitions (subfolders) properly, but it is converting the "=" in the subfolder name to "." and won't let me change it back to a "=" (error message below).  This leads to a corrupt parquet file when someone downloads it. The subdirectory must contain the  "=". I wonder if such an exception can be allowed?

<img width="1174" alt="image" src="https://github.com/user-attachments/assets/375bc680-44b8-433d-9e18-76fea67189e5">

**Comment by pdurbin** [comment]
@kuriwaki that's unfortunate! Would you be able to create a dedicated issue for this (about `=` changing to `.` in the file path) and upload a small parquet file we can test with?

Update: thanks for creating the issue below!

- #10933

**Comment by alungarska** [comment]
Hello! We're looking for a way to exploit parquet's query performance with the [arrow](https://arrow.apache.org/) R package. We want to use the [open_dataset()](https://arrow.apache.org/docs/r/reference/open_dataset.html) function which creates a link to a dataset (as those described in some of the previous posts) without loading it entirely into the memory. This way, we can query the dataset and extract only what's interesting for us. Unfortunately, it seems that we can do this only through the S3 protocol. Is there a way to create an S3 URI to the file(s) of our dataset ? Thanks in advance for any thoughts on the subject!

**Comment by pdurbin** [comment]
@alungarska hi! I think (I hope!) it's possible but I think it would be better to have the discussion at https://dataverse.zulipchat.com or https://groups.google.com/g/dataverse-community (or both!). Would you be able to kick off a thread? And link to it here? Thanks!

**Comment by pdurbin** [comment]
Today I learned that OpenSharing ( https://github.com/OpenSharing-IO/OpenSharing ) supports Parquet. I mentioned this issue in the topic @poikilotherm started: https://dataverse.zulipchat.com/#narrow/channel/432390-large-data/topic/OpenSharing.20at.20Linux.20Foundation/near/602456120
